uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,15 +627,42 @@ define([
       vm.pagination.pageChanged();
     }
 
+    /**
+     * Performs a set of manipulations to prepare all data for calculation and
+     * then triggers the calculation itself if it can be calculated
+     *
+     * @return {Promise}
+     */
+    function attemptCalculateBalanceChange () {
+      updateOpeningBalance();
+      setHoursDeductionsToRequest();
+      setMinMaxDatesToUI();
+      setDatesToRequest();
+      toggleBalance();
+
+      if (!vm.canCalculateChange()) {
+        return $q.resolve();
+      }
+
+      vm.loading.showBalanceChange = true;
+
+      return vm.calculateBalanceChange()
+        .then(setBalanceChange)
```

Comment:
this function it's not covered by the PR, but i have raise some points on it as well. This is the function
```js
function setBalanceChange (balanceChange) {
  if (balanceChange) {
    vm.balance.change = balanceChange;

    calculateOpeningAndClosingBalance();
    rePaginate();
  }

  vm.loading.showBalanceChange = false;
}
```
1. Is it even possible that there is no balance change passed to it? Seems to me a pointless check that should be removed
2. This is not just setting the balance change (which really is just one line of code), but it's also calling `calculateOpeningAndClosingBalance` and `rePaginate` so it's another case of a function that does too much
3. About `calculateOpeningAndClosingBalance`: why are we "calculating" again the opening balance? We just did that before calculating the balance change
4. `vm.loading.showBalanceChange = false;` feels out of place, you would expect `vm.loading.showBalanceChange = true;` to be there at the top of the function, but it's not

Reply:
1. Done. There is really no point.
2,3. calculateOpeningAndClosingBalance is needed there because we need to amend the opening balance if you edit the request, in that case the opening balance is not actually you current balance but your balance before the request has been created. Afair, this was developed by @reneolivo and I think it totally makes sense.
4. This is explained in the comment above, this is because we have no control over the absence period.

Reply:
to be discussed

Reply:
Changed "is" to "was" and now all should be clear. `calculateOpeningAndClosingBalance` is gone.